src_c/base.c

```diff
@@ -557,6 +561,10 @@ pg_UintFromObj(PyObject *obj, Uint32 *val)
         }
```

Comment:
a `PyErr_Clear();` is required here too just before `return 0;` because `PyNumber_Long` sets python error. Actually it would be pretty rare for flow to pass through this codepath because of the initial `PyNumber_Check`, but we still need to clear error for more robust code purposes